.github/workflows/appsignals-e2e-eks-test.yml

```diff
@@ -115,50 +115,53 @@ jobs:
           ${{ env.SAMPLE_APP_NAMESPACE }}
 
       # Application pods need to be restarted for the
-      # app signals instrumentation to take effect
-      - name: Restart the app pods
-        run: kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
-
-      - name: Wait for sample app pods to come up
+      # app signals instrumentation to take effect. After restarting,
+      # check if we can establish connection with the sample app endpoint.
+      - name: Restart the app pods and wait for the endpoints to come online
+        id: restart-pod
         run: |
-          kubectl wait --for=condition=Ready pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+          retry_counter=0
+          max_retry=3
+          success=0
+          while [ $success -eq 0 ]; do
+            if [ $retry_counter -ge $max_retry ]; then
+              exit 1
+            fi
+            kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+            kubectl wait --for=condition=Ready pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+            sample_app_endpoint=http://$(terraform output sample_app_endpoint)
+            if ! $(curl --retry 30 --retry-delay 10 -s -o /dev/null $(echo "$sample_app_endpoint" | tr -d '"'));then
+              echo "Failed to establish connection with endpoint"
+              retry_counter=$(($retry_counter+1))
+            else
+              success=1
+            fi
+          done
```

Comment:
In the worst case, how long will this take? Can we justify taking that long for a single test?